reframe/core/pipeline.py

```diff
@@ -1039,21 +1039,27 @@ def check_performance(self):
             return
 
         with os_ext.change_dir(self._stagedir):
+            # first check and print all values with references
+            value = {}
```

Comment:
Better call this `perf_values`. I'd also change a bit the comment to "Evaluate all performance variables first, so as to log them, then assert them".